src/operations/query/r4ArgsParser.js

```diff
@@ -160,9 +161,28 @@ class R4ArgsParser {
                     })
                 );
             }
+            // same for 'notEquals' parameters
+            if (
+                (notQueryParameterValue && notQueryParameterValue !== '') && (
+                    !Array.isArray(notQueryParameterValue) || notQueryParameterValue.filter(v => v).length > 0
```

Comment:
this will filter out values but only inside the if condition, we should also remove the undefined values while passing it to QueryParameterValue on line 176, filter() function does not update the array instead returns an updated array

Reply:
done